### ISSUE TITLE
Add a dev container feature to install kind

### ISSUE THREAD

**Issue by craiglpeters** [kubernetes-sigs/kind#2967]
**What would you like to be added**:
[Development Containers](https://containers.dev), or dev containers, enable the configuration of remote development environments. A simple json formatted file instructs clients how to construct the container. Clients like VS Code and GitHub Codespaces then use the container to provide developers with a consistent development experience. 

The dev container team maintains a feature that installs [kubectl, helm, and minikube](https://github.com/devcontainers/features/tree/main/src/kubectl-helm-minikube) in the development environment, but there is not feature to install kind for Kubernetes community contributor use.

I propose that we create and distribute a feature for Kubernetes projects that installs kubectl and kind in any dev container. The dev container community supports this model https://github.com/devcontainers/spec/blob/main/proposals/devcontainer-features-distribution.md

**Why is this needed**:
A kubectl-kind feature will simplify the maintenance of any Kubernetes contributor project dev container configuration.

**Comment by craiglpeters** [comment]
I am happy to do the work to create the dev container feature

**Comment by BenTheElder** [comment]
I think I'm missing: What's the ask from KIND? 😅

**Comment by mpriscella** [comment]
This would be great! I've created a similar [feature](https://github.com/mpriscella/features/tree/main/src/kind) to make it easier for my team to install kind in their dev containers.

Prior to me creating that feature, my team and I used minikube by default since the [kubectl, helm, and minikube](https://github.com/devcontainers/features/tree/main/src/kubectl-helm-minikube) feature that @craiglpeters mentioned was _much_ easier to install across different dev containers. I believe most engineers working on a Kubernetes project would opt to use kind if it was a feature.

**Comment by craiglpeters** [comment]
@BenTheElder - @mpriscella addressed the user value.

The ask of the KIND community is to create and maintain the feature following this [template](https://github.com/devcontainers/feature-template) so that all devs using dev containers can easily add kind to their configurations. I am happy to step up as a community member to create and maintain the feature. It needs to live in its own repo. I propose that the community create a new repo called /kubernetes-sigs/devcontainer-features where features for multiple tools used by the community could be maintained, kind first among them.

**Comment by BenTheElder** [comment]
I see.

We cannot create a repo ourselves, the proposal to create a repo has to be brought to a SIG.

if we can stick multiple in a repo, is there a reason we can't put one next to https://github.com/devcontainers/features/tree/main/src/kubectl-helm-minikube ?

**Comment by craiglpeters** [comment]
The model the dev container community is promoting is for tools to publish and maintain their own features. I am happy to take this to a SIG to discuss. I'll bring this up with SIG Testing next week in person
cc: @dims

**Comment by BenTheElder** [comment]
I don't know if SIG Testing would be the right owner for a shared devcontainers repo, but also we met today, so the next meeting will be in two weeks.

I think we should probably unify the discussion with https://github.com/kubernetes/kubernetes/issues/113019

**Comment by dims** [comment]
we had one repo already for something similar, we could repurpose/rename?

**Comment by craiglpeters** [comment]
> we had one repo already for something similar, we could repurpose/rename?

Absolutely

**Comment by devinsaini** [comment]
@mpriscella I used the kind feature and it sets up kind in the devcontainer fine. However, I had some issues starting a kind cluster. Is it possible to use it with docker-from-docker setup?
I either get /etc/kubernetes/admin.conf not found, which I copied manually but still get the error,
or I get The connection to the server test-control-plane:6443 was refused. 
I can see it tries to start a kind node in docker desktop but the node eventually shuts down. My host is Windows.

**Comment by macel94** [comment]
any news? 

<img width="498" height="480" alt="Image" src="https://github.com/user-attachments/assets/a46dbfa9-3ba6-4345-b8af-bc2ea1883539" />

**Comment by BenTheElder** [comment]
No. And we really don't have bandwidth to maintain more things.

I haven't seen further demand for this, nor has there been activity since 2022.

**Comment by joebowbeer** [comment]
Fortunately, there are two features in the registry that install `kind`:

https://containers.dev/features?search=Kind

My [arkade-get](https://github.com/joebowbeer/devcontainers-features/tree/main/src/arkade-get) feature can also install `kind`.
